enhancements/machine-config/certificate-authorities-for-image-registries.md

```diff
@@ -0,0 +1,185 @@
+image registry operator is not installed.
+
+## Motivation
+
+The image registry operator manages a daemon set that distributes certificate
+authorities for the integrated image registry, and it also distributes
+certificate authorities for external registries that are provided by the
+cluster administrator in the image config. This makes the image registry
+operator required for the cluster while the integrated image registry is
+optional and can be removed.
+
+The goal of this enhancement is to move this required functionality to an
+essential cluster operator, and have the image regsitry operator only manage
+the optional registry workload.
+
+### User Stories
+
+* As a <role>, I want to <take some action> so that I can <accomplish a
+goal>.
+
+### Goals
+
+* Make additionalTrustedCA be managed by machine-config-operator.
+* Create a mechanism for the cluster-image-registry-operator to distribute
+  certificate authorities for the integrated image registry.
+* Remove the node-ca daemon set.
+
+### Non-Goals
+
+* Change additionalTrustedCA behavior for its users.
+
+## Proposal
+
+Create a controller inside cluster-config-operator that will replace the
+node-ca daemon set.
+
+This controller should handle and distribute to all nodes certificate
+authorities from the user-provided config map that is specified in
+`images.config.openshift.io/cluster`. It should also observe a config map in
+the `openshift-config-managed` namespace and merge its content with the
+user-provided config map. This will allow cluster-image-registry-operator to
+provide certificate authorities for the integated image registry without
+changing the user-provided config map.
+
+Once this controller is created, the node-ca daemon set should be removed.
+
+### Workflow Description
+
+Let's suppose the user creates a config map with the certificate authorities
+for their registries:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: registry-ca
+  namespace: openshift-config
+data:
+  registry.example.com: |
+    -----BEGIN CERTIFICATE-----
+    ...
+    -----END CERTIFICATE-----
+  registry-with-port.example.com..5000:
+    -----BEGIN CERTIFICATE-----
+    ...
+    -----END CERTIFICATE-----
+```
+
+Then updates the `images.config.openshift.io/cluster` object to point to this
+config map:
+
+```yaml
+spec:
+  additionalTrustedCA:
+    name: registry-ca
+```
+
+If the image registry operator is installed, there should be a config map:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: image-registry-ca
+  namespace: openshift-config-managed
+data:
+  image-registry.openshift-image-registry.svc..5000: |
+    -----BEGIN CERTIFICATE-----
+    ...
+    -----END CERTIFICATE-----
+  image-registry.openshift-image-registry.svc.cluster.local..5000: |
+    -----BEGIN CERTIFICATE-----
+    ...
+    -----END CERTIFICATE-----
+```
+
+The cluster-config-operator should update every node to propagete these
```

Comment:
nit: "propagete" -> propagate"

Reply:
the cluster-config-operator is a dead repo.  It contains no code and should contain no code.  It has no owners.  It holds CRD and CR definitions required to bootstrap the first master, nothing more.

Reply:
#812 had floated a node operator.  Allowing folks to save some CPU/memory by removing the registry operator, while replacing it with a new node operator, doesn't actually win many resources back for the customer.  But #812 shows that we have other reasons for wanting the node folks to have their own operator, including avoiding the need to do things like openshift/machine-config-operator#1453.  No worries from me if the image-registry folks can find a team with an existing core operator to hand this functionality off to.  Or if the node folks can continue to talk other existing core operators into letting them co-maintain node-associated portions.  But I'm cross-linking #812 in case folks decide to go that route instead.

Reply:
I meant machine-config-operator, it already manages files like `registries.conf`. Updated.